pkg/planner/core/casetest/physicalplantest/testdata/plan_suite_in.json

```diff
@@ -354,6 +354,16 @@
       "select max(a) from (select t1.a from t t1 join t t2 on t1.a=t2.a) t"
     ]
   },
+  {
+    "name": "TestIssue45822",
+    "cases": [
+      "select count(*) from (select a,(select t2.b from t t2,t t3 where t2.a=t3.a and t1.a=t2.a limit 1) t from t t1) t",
```

Comment:
Please add more test
1. subquery not in the projection such as where subquery, from (subquery).  Make sure that subqueries elsewhere than projection are not affected by this optimization.
2. subquery nested in multiple query block layers